homeassistant/components/sensor/forecast.py

```diff
@@ -124,78 +116,78 @@ def state(self):
-            elif self.type == 'precip_probability':
-                self._state = round(data.precipProbability * 100, 1)
-            elif self.type == 'dew_point':
-                self._state = round(data.dewPoint, 1)
-            elif self.type == 'temperature':
-                self._state = round(data.temperature, 1)
-            elif self.type == 'apparent_temperature':
-                self._state = round(data.apparentTemperature, 1)
-            elif self.type == 'wind_speed':
-                self._state = data.windSpeed
-            elif self.type == 'wind_bearing':
-                self._state = data.windBearing
-            elif self.type == 'cloud_cover':
-                self._state = round(data.cloudCover * 100, 1)
-            elif self.type == 'humidity':
-                self._state = round(data.humidity * 100, 1)
-            elif self.type == 'pressure':
-                self._state = round(data.pressure, 1)
-            elif self.type == 'visibility':
-                self._state = data.visibility
-            elif self.type == 'ozone':
-                self._state = round(data.ozone, 1)
-
+            else:
+                if self.type == 'summary':
+                    self._state = data.summary
+                elif self.type == 'icon':
+                    self._state = data.icon
+                elif self.type == 'nearest_storm_distance':
+                    self._state = data.nearestStormDistance
+                elif self.type == 'nearest_storm_bearing':
+                    self._state = data.nearestStormBearing
+                elif self.type == 'precip_intensity':
+                    self._state = data.precipIntensity
+                elif self.type == 'precip_type':
+                    self._state = data.precipType
+                elif self.type == 'precip_probability':
+                    self._state = round(data.precipProbability * 100, 1)
+                elif self.type == 'dew_point':
+                    self._state = round(data.dewPoint, 1)
+                elif self.type == 'temperature':
+                    self._state = round(data.temperature, 1)
+                elif self.type == 'apparent_temperature':
+                    self._state = round(data.apparentTemperature, 1)
+                elif self.type == 'wind_speed':
+                    self._state = data.windSpeed
+                elif self.type == 'wind_bearing':
+                    self._state = data.windBearing
+                elif self.type == 'cloud_cover':
+                    self._state = round(data.cloudCover * 100, 1)
+                elif self.type == 'humidity':
+                    self._state = round(data.humidity * 100, 1)
+                elif self.type == 'pressure':
+                    self._state = round(data.pressure, 1)
+                elif self.type == 'visibility':
+                    self._state = data.visibility
+                elif self.type == 'ozone':
+                    self._state = round(data.ozone, 1)
```

Comment:
@fabaff  @MartinHjelmare I'm now thinking a psudo-switch isn't a good idea for this.

the object here, `data`, is an instance of a class we get from the forecast SDK we are using. When we call a property on it, a method is executed that checks to see if the json object response from the API included that attribute. If the attribute is missing, it raises a `PropertyUnavailable` exception. Hence the `try/catch` wrapped around the `if/elif/elif/elif...`. (So far this seems to sometime occur for `data.precipType`.)

To do a proper switch, each case would have to be function that catches the above error. Otherwise, a single missing property with break ALL of the lookups during dict creation, not just the failed one. 

Going down that path, it started to be a decent amount of code and.....frankly....the long and ugly `if/elif/elif/elif....` is much easier to read and much more straightforward to understand.

Unless someone has an idea of a clever switch I might not have considered, I'm thinking it's best to leave it as is.